### ISSUE TITLE
Style checks aren't included in the VSEmulator composition tool output

### ISSUE THREAD

**Issue by mrlacey** [microsoft/WindowsTemplateStudio#3229]
Debugging issues connected to style checks and composition is tricky.
This is made harder due to a lack of support from the Composition Tool.

In VSEmulator, when specifying that style checks should be included

![image](https://user-images.githubusercontent.com/189547/60603245-4e4bbc00-9dad-11e9-95c9-a524d6a232b6.png)

The added composition templates that add the style checks aren't included in the composition tool output.
This means it can't be used to debug issues where style checks aren't being included in tests of new templates.

I'd like to see all templates included in what's displayed in the composition tool.

**Comment by sibille** [comment]
@mvegaca, can we show style check templates in the composition tool?

**Comment by sibille** [comment]
@mrlacey, spoke to @mvegaca, this is not possible right now, as the stylecop templates are added from the emulator code before generation.

**Comment by mrlacey** [comment]
This sounds like a reason why they're not showing up at the moment, not a reason why they couldn't be made to be included.

**Comment by sibille** [comment]
We could make them available as feature, then they would show up, but I think we decided not to do so as we can't always guarantee all rules are met.

**Comment by mrlacey** [comment]
Everything that's done for these style templates is a special feature just for handling them.
There should be no need to treat them like other templates just to get them to show up.

> we can't always guarantee all rules are met.

The very reason for the existence of these templates is to ensure that the generated code always meets these rules. This is a non-argument. 
My understanding was that the reason was that these templates were made available in the vsemulator was to make it easier to manually test that the code meets the required rules.
These rules are enforced in automated tests. The templates must meet these rules.

If it's a lot of work and therefore isn't deemed worth the effort required then that's an understandable reason.
I don't believe that this is something that isn't possible though.


The option was added to make it easier to check that templates meet the required formatting rules.
That they only work with part of the functionality of VSEmulator was confusing and wasted a lot of my time. That is why I opened this issue. I'll just remember not to use this in future and keep using the automated tests to verify generated code meets the required standards.

**Comment by sibille** [comment]
The Stylecop templates are very useful and have provided a lot of value to the project through the tests. 

Having them in the emulator is reducing template construction time a lot for us. I'm sorry the issue you had with the composition wasted time, this was surely not our intent and I agree that it would be great to have them show up in the composition tool to be able to diagnose failures easier. 

We added them in the emulator to not add code that is only used for tests/emulation to the wizard. Turning them into a feature would solve both questions + I think it can be useful for people to keep enforcing the same style rules the project was set up with, thats why I proposed it.

**Comment by mrlacey** [comment]
Add the StyleCop rules we use as a feature would likely create the expectation that we (and by extension, Microsoft) recommend the settings provided. 
In reality, the settings we test against are as much about ensuring consistency in what we produce as they are about ensuring good practices.
If I was adding StyleCop rules to an app generated with WinTS I would want to remove most of the suppressions we include and address those issues that are currently excluded because they "would add unnecessary complexity to template composition."

Attempting to include stylecop templates that wouldn't need suppressions to make template authoring easier would require complex post actions to ensure the generated code meets basic rules for ordering. These in turn would mean that right-click actions could not do the necessary merging without, potentially making other changes to the code base that would have to allow for other modifications made by the developer,.


As has been discussed several times previously, the current styling consistency checks are designed to be a balance between ensuring consistency in the generated code without making template authoring unnecessarily complicated. They were never intended to be something that was shipped and have not been created for that purpose.



Also note that for the VB templates we're also running some checks/analyzers that I wrote to ensure the consistency that we require. If we included these in the generated code there would be (I estimate) considerable work involved in getting them to a sufficient standard for public consumption.
